src/app/Library/CrudPanel/CrudColumn.php

```diff
@@ -78,10 +78,16 @@ public function key(string $key)
         if (! isset($this->attributes['name'])) {
             abort(500, 'Column name must be defined before changing the key.');
         }
-        if ($this->crud()->hasColumnWhere('key', $this->attributes['key'])) {
-            $this->crud()->setColumnDetails($this->attributes['key'], array_merge($this->attributes, ['key' => $key]));
+
+        $columns = $this->crud()->columns();
+        $searchKey = $this->attributes['key'];
+
+        if (isset($columns[$searchKey])) {
+            unset($columns[$searchKey]);
+            $column['key'] = $key;
         }
-        $this->attributes['key'] = $key;
+        $this->attributes = $column;
+        $this->setOperationSetting('columns', array_merge($columns, [$key => $column]));
```

Comment:
Guys, are you sure this is working? 

My spidey-sense got triggered because I see we're referencing both `$column` and `$columns` variables, but `$column` (singular) is only created inside a conditional, from what I can tell. That leaves open a scenario where `$column` is not defined, correct?

Maybe there's a missing `$column = $this->attributes;` at the top, or something?

Reply:
Yes, this works, but not 100% as expected. 

Like you said, it has corner cases where it would fail and what stood out to me was the fact that in @jorgetwgroup screenshot the columns are not displaying the the same value, as they should since they are all the same column.

I've fixed it with your spot on suggestion!

![image](https://user-images.githubusercontent.com/7188159/183379035-b5308352-1a5c-402d-a3be-25fd9ea41df3.png)

Reply:
Awesome. Back to @jorgetwgroup for a re-check then.

Reply:
(I know this is a minor issue to do all this back-and-forth on. but it's good we get into the habit of it)